andela/mervels-bn-backend
# #168131423 Notifications for new travel request

#### What does this PR do?
Email and in-App notifications for new travel request
#### Description of Task to be completed?
- Implement notifications so that it notifies the manager when a new travel request is initiated
- Manager can opt-out or opt-in to receiving the email notifications
#### How should this be manually tested?
- Open `https://bn-notifications.herokuapp.com` in the browser and allow notifications
- Make a POST request through postman to the review app `https://mervels-bn-backend-stagi-pr-46.herokuapp.com/api/v1/requests/one-way` with the payload
```
{
  "from": "Kigali, Rwanda",
  "to": 1,
  "travelDate": "2020-10-10",
  "reason": "iam travelling cause the company allows us to, i mean the company finances everything so why not",
  "accommodation": "hotel"
}
```
and you should see a notification upon request success
- To test the above feature you can get info on the setup from this [repo](https://github.com/drayzii/notifications) being used for notifications
- for opting-in ot opting-out
PATCH `/api/v1/auth/email-preferences` for both opting in and opting out. you need to provide the token
- check the manager inbox `marveldev53@gmail;.com`, you should receive an email similar to the one in the screenshot when someone requests a trip
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168131423](https://www.pivotaltracker.com/story/show/168131423)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/50402526/65983947-f6d9ec80-e47e-11e9-8352-b6c1eecd75e6.png)
![image](https://user-images.githubusercontent.com/39488016/65985418-0d357780-e482-11e9-8a37-4afb60bee62f.png)
![image](https://user-images.githubusercontent.com/39488016/65985713-a795bb00-e482-11e9-9387-0317102950a0.png)
![image](https://user-images.githubusercontent.com/39488016/65985737-b2505000-e482-11e9-9fdd-3361c081d49a.png)


#### Questions:
N/A
